api/src/main/java/org/openmrs/module/queue/api/impl/QueueEntryServiceImpl.java

```diff
@@ -225,4 +250,18 @@ private void endQueueEntry(@NotNull QueueEntry queueEntry) {
 		queueEntry.setEndedAt(new Date());
 		dao.createOrUpdate(queueEntry);
 	}
+	
+	@Override
+	public QueueEntry getPreviousQueueEntry(@NotNull QueueEntry queueEntry) {
```

Comment:
This needs to be marked as @Transactional(readOnly=true) to indicate this is a read-only operation